### PR TITLE
Improve error message for logging test function

### PR DESCRIPTION
**Pull request by aiwantaozi**
Improve logging test function return error message

Problem:
Error message for logging test related function will return to UI, need more clear

Solution:
improve error message

Issue:
https://github.com/rancher/rancher/issues/18546

### CODE REVIEW

**Comment by orangedeng** [comment]
LGTM

File: pkg/controllers/user/logging/utils/kafka.go [hunk 1]
```diff
@@ -134,3 +135,11 @@ func (w *kafkaTestWrap) kafkaConn(dial dialer.Dialer, config *tls.Config, smartH
 
 	return kafka.NewConn(LeaderConn, w.Topic, defaultPartition), nil
 }
+
+func wrapErrEOF(err error) error {
+	if err == io.EOF {
+		return errors.New("unexpected EOF, connect is closed by remote server")
```
Comment (cjellick) [inline pkg/controllers/user/logging/utils/kafka.go]: connection, not connect

File: pkg/controllers/user/logging/utils/kafka.go [hunk 2]
```diff
@@ -103,13 +104,13 @@ func (w *kafkaTestWrap) kafkaConn(dial dialer.Dialer, config *tls.Config, smartH
 
 	if err := kafkaConn.CreateTopics(topicConf); err != nil {
 		kafkaConn.Close()
-		return nil, err
+		return nil, errors.Wrapf(wrapErrEOF(err), "create kafka topic %s failed", w.Topic)
```
Comment (cjellick) [inline pkg/controllers/user/logging/utils/kafka.go]: doesnt sound right. should be
"creation of kafka topic %s failed"

File: pkg/controllers/user/logging/utils/kafka.go [hunk 3]
```diff
@@ -103,13 +104,13 @@ func (w *kafkaTestWrap) kafkaConn(dial dialer.Dialer, config *tls.Config, smartH
...
 	}
 
 	partitions, err := kafkaConn.ReadPartitions(w.Topic)
 	if err != nil {
 		kafkaConn.Close()
-		return nil, err
+		return nil, errors.Wrap(wrapErrEOF(err), "read kafka patitions failed")
```
Comment (cjellick) [inline pkg/controllers/user/logging/utils/kafka.go]: "couldn't read kafka partions" might be better

File: pkg/controllers/user/logging/utils/kafka.go (hunk 2 above)
Comment (cjellick) [reply pkg/controllers/user/logging/utils/kafka.go]: or "couldn't create kafka topic %s"

File: pkg/controllers/user/logging/utils/kafka.go (hunk 1 above)
Comment (cjellick) [reply pkg/controllers/user/logging/utils/kafka.go]: and the tense isn't quite right, dropping "is" might make it sound better
"connection closed by remote server"

File: pkg/controllers/user/logging/utils/testwrap.go [hunk 4]
```diff
@@ -98,7 +98,7 @@ func writeToUDPConn(data []byte, smartHost string) error {
 func newTCPConn(dialer dialer.Dialer, smartHost string, tlsConfig *tls.Config, handshake bool) (net.Conn, error) {
 	conn, err := dialer("tcp", smartHost)
 	if err != nil {
-		return nil, errors.Wrap(err, "create raw conn failed")
+		return nil, errors.Wrapf(err, "create raw conn to %s failed", smartHost)
```
Comment (cjellick) [inline pkg/controllers/user/logging/utils/testwrap.go]: "couldn't create raw connection %s"

**Comment by aiwantaozi** [comment]
@cjellick thanks, the pr is updated

File: pkg/controllers/user/logging/utils/splunk.go [hunk 5]
```diff
@@ -24,15 +24,15 @@ type splunkTestWrap struct {
...
+		return errors.Wrapf(err, "couldn't parse url %s", w.Endpoint)
 	}
 
 	isTLS := url.Scheme == "https"
 	var tlsConfig *tls.Config
 	if isTLS {
 		tlsConfig, err = buildTLSConfig(w.Certificate, w.ClientCert, w.ClientKey, w.ClientKeyPass, "", url.Hostname(), w.SSLVerify)
 		if err != nil {
-			return errors.Wrap(err, "build tls config failed")
+			return err
```
Comment (orangedeng) [inline pkg/controllers/user/logging/utils/splunk.go]: Please explain a little why it doesn’t require error wrapping here.

Comment (aiwantaozi) [reply pkg/controllers/user/logging/utils/splunk.go]: because error is from buildTLSConfig, we already wrap error inside buildTLSConfig function

**Comment by cjellick** [comment]
@orangedeng 
> I would prefer using fail to do something instead of couldn’t do something. It seems a bit more formal.

you are probably correct. I don't care as much about the exact wording as long as it is grammatically correct

**Comment by cjellick** [comment]
lgtm

**Pull request merged**